CRM/Core/BAO/File.php

```diff
@@ -689,10 +689,31 @@ public static function deleteAttachment() {
       //     ],
       //   ]);
       // TEMP HACK
-      CRM_Core_DAO::executeQuery("UPDATE `{$customGroup['table_name']}` SET {$customField['column_name']} = NULL WHERE entity_id = %1 AND {$customField['column_name']} = %2", [
+      $ids = CRM_Core_DAO::executeQuery("SELECT id FROM `{$customGroup['table_name']}` WHERE entity_id = %1 and {$customField['column_name']} = %2", [
         1 => [$params['entityID'], 'Integer'],
         2 => [$params['fileID'], 'Integer'],
       ]);
+      $customParams = [$customGroup['table_name'] => []];
+      while ($ids->fetch()) {
+        $customParams[$customGroup['table_name']][$dao->id] = [];
+        $customParams[$customGroup['table_name']][$dao->id][] = [
+          'custom_group_id' => $customGroup['id'],
+          'custom_field_id' => $customField['id'],
+          'entity_id' => $params['entityID'],
+          'entityTable' => $params['entityTable'],
+          'file_id' => FALSE,
+          'table_name' => $customGroup['table_name'],
+          'id' => $ids->id,
+          'is_multiple' => $customGrouo['is_multiple'],
+          'serialize' => 0,
+          'type' => 'File',
+          'value' => FALSE,
+          'table_name' => $customGroup['table_name'],
+          'column_name' => $customField['column_name'],
+        ];
+      }
+      var_dump($customParams);
```

Comment:
var_dump

Also did you see https://github.com/civicrm/civicrm-core/pull/32733? Related but different scope.

Reply:
I hadn't looked at that but will, and I'll remove that var_dump shortly